File: src/ProjectR/content/scripts/spells/RecklessSwing.lua
function GetTargetType()
    return Single
end

function GetName()
return "Reckless Swing"
end

function GetDescription()
return "Strike recklessly at your foe.\nAlso damages self slightly."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .66
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 6.5 * aAD - 1.125 * dDEF
  local selfDamage = .16 * aAD - .2 * aDEF
  a:TakeDamage(selfDamage)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/RecklessSwing.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class RecklessSwing : SpellScriptBase
    {
        public override string Name { get { return "Reckless Swing"; } }
        public override string Description { get { return "Strike recklessly at your foe.\nAlso damages self slightly."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 66; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 6.5 * aAD - 1.125 * dDEF;
            var selfDamage = .16 * aAD - .2 * aDEF;
            DealDamage(caster, selfDamage);
            DealDamage(damage);
        }
    }
}